python/pyspark/ml/param/_shared_params_code_gen.py

```diff
@@ -115,7 +118,7 @@ def get$Name(self):
         ("outputCol", "output column name", None),
         ("numFeatures", "number of features", None),
         ("checkpointInterval", "checkpoint interval (>= 1)", None),
-        ("seed", "random seed", None),
+        ("seed", "random seed", "random.randint(0, sys.maxsize)"),
```

Comment:
I agree that setting the default seed to 42 globally is quite funny. But I think it is nice to have a fixed seed for each component, so it is easy to reproduce the result. For example, could you send the seed based on the hash value of the class name?